executor/linux/step.go

```diff
@@ -438,3 +451,39 @@ func (c *client) DestroyStep(ctx context.Context, ctn *pipeline.Container) error
 
 	return nil
 }
+
+// getSecretValues is a helper function that creates a slice of
+// secret values that will be used to mask secrets in logs before
+// updating the database.
+func getSecretValues(ctn *pipeline.Container) []string {
+	secretValues := []string{}
+	// gather secrets' values from the environment map for masking
+	for _, secret := range ctn.Secrets {
+		s := ctn.Environment[strings.ToUpper(secret.Target)]
+		// handle multi line secrets from files
+		s = strings.ReplaceAll(s, "\n", " ")
+
+		// drop any trailing spaces
+		if strings.HasSuffix(s, " ") {
+			s = s[:(len(s) - 1)]
+		}
+		secretValues = append(secretValues, s)
+	}
+	return secretValues
+}
+
+// maskSecrets is a helper function that takes in a byte array
+// and a slice of secret values to mask.
+func maskSecrets(log []byte, secrets []string) []byte {
+	strData := string(log)
+	for _, secret := range secrets {
+		re := regexp.MustCompile(`\s` + secret + `\s`)
+		matches := re.FindAllString(strData, -1)
+		for _, match := range matches {
+			mask := string(match[0]) + constants.SecretLogMask + string(match[len(match)-1])
+			strData = strings.Replace(strData, match, mask, 1)
+		}
+		strData = re.ReplaceAllString(strData, constants.SecretLogMask)
+	}
+	return []byte(strData)
+}
```

Comment:
Do we think this should be a method on the `github.com/go-vela/library/Log{}` type?

Thinking the function signature could look like:

```go
func (l *Log) MaskData(secrets []string) {}
```

Then, to use this function in the the code above:

```go
// update the existing log with the new bytes
//
// https://pkg.go.dev/github.com/go-vela/types/library?tab=doc#Log.AppendData
_log.AppendData(logs.Bytes())

// mask any secrets in the log data
//
// https://pkg.go.dev/github.com/go-vela/types/library?tab=doc#Log.MaskData
_log.MaskData(secretValues)
```

Reply:
That might be helpful in the eventual effort to have this masking apply to external secrets

Reply:
You read my mind 👍  I was thinking along those same lines!

Also, if/when we write out other executors then we don't have to duplicate code.

In the interest of not blocking the functionality, I'll approve for now 😄